### ISSUE TITLE
In dash ${@} parameter transformation is undefined

### ISSUE THREAD

**Issue by tdukv** [koalaman/shellcheck#2782]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0 and online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

var="SoMe tExT"

printf "%s\n" "${var@L}"
printf "%s\n" "${var@U}"

```

#### Here's what shellcheck currently says:
Nothing.

#### Here's what I wanted or expected to see:
Similar to SC3059 about ${,,} or ${^^} modification.

**Comment by aeiplatform** [comment]
What is this substitution?
Even Bash says `bad substitution`.

**Comment by Ordoviz** [comment]
@aeiplatform See `${parameter@operator}` in [3.5.3 Shell Parameter Expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html).

**Comment by aeiplatform** [comment]
You are right, I'm not sure why that bad substitution happened, but a the latest bash version this works.
